packages/x-charts/src/BarChart/BarChart.tsx

```diff
@@ -57,8 +57,8 @@ export function BarChart(props: BarChartProps) {
         tooltip?.trigger !== 'axis' && highlight?.x === 'none' && highlight?.y === 'none'
       }
     >
-      <BarPlot />
       <Axis topAxis={topAxis} leftAxis={leftAxis} rightAxis={rightAxis} bottomAxis={bottomAxis} />
+      <BarPlot />
```

Comment:
What is the impact of this change ?
Does it change the layering of the chart ?

Reply:
Yes, elements are rendered in their order. It's mostly to get line dots on top of axes instead of the opposite.

Later on, plots should have a clipping option to prevent them from to overflow the area defined by axes.